src/net/ossl.cc

```diff
@@ -775,6 +843,18 @@ class session : public enable_shared_from_this<session>, public session_impl {
             }
             SSL_set_connect_state(_ssl.get());
         }
+
+        if (_type == session_type::CLIENT && !_options.session_resume_data.empty()) {
+            auto data_ptr = reinterpret_cast<unsigned char *>(_options.session_resume_data.data());
+            auto sess = ssl_session_ptr(d2i_SSL_SESSION(nullptr, const_cast<const unsigned char **>(&data_ptr), _options.session_resume_data.size()));
```

Comment:
nit: I think you can just do this instead (`const_cast` set off an alarm in my head). `_options.session_resume_data.data()` already returns `unsigned char*` and you can add a `const` without `const_cast`.
```suggestion
            const unsigned char* data_ptr = _options.session_resume_data.data();
            auto sess = ssl_session_ptr(d2i_SSL_SESSION(nullptr, &data_ptr, _options.session_resume_data.size()));
            
```

Reply:
very nice